Mathlib/Analysis/CStarAlgebra/ContinuousFunctionalCalculus/Isometric.lean

```diff
@@ -0,0 +1,488 @@
+    IsometricContinuousFunctionalCalculus ℂ A IsStarNormal where
+  isometric a ha := by
+    rw [cfcHom_eq_of_isStarNormal]
+    exact isometry_subtype_coe.comp <| StarAlgEquiv.isometry (continuousFunctionalCalculus a)
+
+instance IsSelfAdjoint.instIsometricContinuousFunctionalCalculus :
+    IsometricContinuousFunctionalCalculus ℝ A IsSelfAdjoint :=
+  SpectrumRestricts.isometric_cfc Complex.reCLM Complex.isometry_ofReal (.zero _)
+    fun _ ↦ isSelfAdjoint_iff_isStarNormal_and_spectrumRestricts
+
+end Complex
+
+section NNReal
+
+variable {A : Type*} [NormedRing A] [PartialOrder A] [StarRing A] [StarOrderedRing A]
+variable [NormedAlgebra ℝ A] [IsometricContinuousFunctionalCalculus ℝ A IsSelfAdjoint]
+variable [NonnegSpectrumClass ℝ A] [UniqueContinuousFunctionalCalculus ℝ A]
+
+open NNReal in
+instance Nonneg.instIsometricContinuousFunctionalCalculus :
+    IsometricContinuousFunctionalCalculus ℝ≥0 A (0 ≤ ·) :=
+  SpectrumRestricts.isometric_cfc (q := IsSelfAdjoint) ContinuousMap.realToNNReal
+    isometry_subtype_coe le_rfl (fun _ ↦ nonneg_iff_isSelfAdjoint_and_spectrumRestricts)
+
+end NNReal
+
+end Unital
+
+section NonUnital
+
+section Complex
+
+variable {A : Type*} [NonUnitalCStarAlgebra A]
+
+open Unitization
+
+
+open ContinuousMapZero in
+instance IsStarNormal.instNonUnitalIsometricContinuousFunctionalCalculus :
+    NonUnitalIsometricContinuousFunctionalCalculus ℂ A IsStarNormal where
+  isometric a ha := by
+    refine AddMonoidHomClass.isometry_of_norm _ fun f ↦ ?_
+    rw [← norm_inr (𝕜 := ℂ), ← inrNonUnitalStarAlgHom_apply, ← NonUnitalStarAlgHom.comp_apply,
+      inr_comp_cfcₙHom_eq_cfcₙAux a, cfcₙAux]
+    simp only [NonUnitalStarAlgHom.comp_assoc, NonUnitalStarAlgHom.comp_apply,
+      toContinuousMapHom_apply, NonUnitalStarAlgHom.coe_coe]
+    rw [norm_cfcHom (a : Unitization ℂ A), StarAlgEquiv.norm_map]
+    rfl
+
+instance IsSelfAdjoint.instNonUnitalIsometricContinuousFunctionalCalculus :
+    NonUnitalIsometricContinuousFunctionalCalculus ℝ A IsSelfAdjoint :=
+  QuasispectrumRestricts.isometric_cfc Complex.reCLM Complex.isometry_ofReal (.zero _)
+    fun _ ↦ isSelfAdjoint_iff_isStarNormal_and_quasispectrumRestricts
+
+end Complex
+
+section NNReal
+
+variable {A : Type*} [NonUnitalNormedRing A] [PartialOrder A] [StarRing A] [StarOrderedRing A]
+variable [NormedSpace ℝ A] [IsScalarTower ℝ A A] [SMulCommClass ℝ A A]
+variable [NonUnitalIsometricContinuousFunctionalCalculus ℝ A IsSelfAdjoint]
+variable [NonnegSpectrumClass ℝ A] [UniqueNonUnitalContinuousFunctionalCalculus ℝ A]
+
+open NNReal in
+instance Nonneg.instNonUnitalIsometricContinuousFunctionalCalculus :
+    NonUnitalIsometricContinuousFunctionalCalculus ℝ≥0 A (0 ≤ ·) :=
+  QuasispectrumRestricts.isometric_cfc (q := IsSelfAdjoint) ContinuousMap.realToNNReal
+    isometry_subtype_coe le_rfl (fun _ ↦ nonneg_iff_isSelfAdjoint_and_quasispectrumRestricts)
+
+end NNReal
+
+end NonUnital
+
+end Instances
+
+/-! ### Properties specific to `ℝ≥0` -/
+
```

Comment:
Do we also want versions for `ℝ` (i.e. `f x` ≤ ‖cfc f a‖`, etc)?

Reply:
I think that's not necessary, as you can just `trans` with `Real.le_norm_self` or something.